forloop_modules/function_handlers/webscraping_handlers.py

```diff
@@ -2593,6 +2593,159 @@ def export_imports(self, *args):
+        write_mode = node_detail_form.get_chosen_value_by_name("write_mode", variable_handler)
+        new_var_name = node_detail_form.get_chosen_value_by_name("new_var_name", variable_handler)
+
+        new_var_name = self.update_node_fields_with_shown_dataframe(node_detail_form, new_var_name)
+        self.direct_execute(xpaths, entry_df, write_mode, columns)
+        ncrb.update_last_active_dataframe_node_uid(node_detail_form.node_uid)
+
+    def execute_with_params(self, params):
+        xpaths = params["xpaths"]
+        columns = params["columns"]
+        entry_df = params["entry_df"]
+        write_mode = params["write_mode"]
+        new_var_name = params["new_var_name"]
+
+        self.direct_execute(xpaths, columns, entry_df, write_mode, new_var_name)
+
+    def direct_execute(self, xpaths, columns, entry_df, write_mode, new_var_name):
+        def is_list_of_strings(var) -> bool:
+            return isinstance(var, list) and not all(isinstance(v, str) for v in var)
+
+        if isinstance(xpaths, str) or is_list_of_strings(xpaths):
+            raise CriticalPipelineError("XPaths must be provided as a string or a list of strings")
+        if isinstance(columns, str) or is_list_of_strings(columns):
+            raise CriticalPipelineError("Columns must be provided as a string or as a list of strings")
+        if len(columns) != len(xpaths):
+            raise CriticalPipelineError("Selected XPaths must be of same length as Columns")
+
+        inp = Input()
+        inp.assign("xpaths", xpaths)
+        inp.assign("columns", columns)
+        inp.assign("entry_df", entry_df)
+        inp.assign("write_mode", write_mode)
+        inp.assign("new_var_name", new_var_name)
+
+        filename = f'{new_var_name}.txt'
+        data_dict = {}
+        for xpath, column in zip(xpaths, columns):
+            xpath = suh.check_xpath_apostrophes(xpath)
+            suh.webscraping_client.extract_xpath(xpath, filename, "w+")
+            data = suh.wait_until_data_is_extracted(filename, timeout=3, xpath_func=True)
+            if isinstance(data, list):
+                data_dict[column] = data
+            elif isinstance(data, str):
+                data_dict[column] = [data]
+
+        if write_mode == "Write":
+            new_df = pd.DataFrame(data_dict)
+        elif write_mode == "Append":
+            if entry_df is None:
+                new_df = pd.DataFrame(data_dict)
+            else:
+                new_rows = pd.DataFrame(data_dict)
+                new_df = pd.concat([entry_df, new_rows], ignore_index=True)
+
+        if new_var_name in variable_handler.variables.keys():
+            variable_handler.update_variable(new_var_name, new_df)
+        else:
+            variable_handler.create_variable(new_var_name, new_df)
+
+    def input_execute(self, inp: Input):
+        raise NotImplementedError()
+
+    def export_code(self, node_detail_form):
```

Comment:
Again, better to not have it here at all. Two main reasons here:

1. There are checks in code inspect functionality that will return some generic comment like this in case of a failure of building the code. Moreover this comment would be confusing as it would differ from others which failed to build for some reason.
2. Same as above, we won't see this in tests. If you delete `input_execute` and `export_code` the test will warn us and we will know that this handler is incomplete and needs some future work.